src/escrows/VaultHelper.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {IERC4626} from "lib/openzeppelin-contracts/contracts/interfaces/IERC4626.sol";
+import {IERC20} from "lib/openzeppelin-contracts/contracts/interfaces/IERC20.sol";
+import {IMarket} from "src/interfaces/IMarket.sol";
+import {SafeERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+@title Vault Helper
+@notice This contract is a helper contract for an ERC4626 vault contract. 
+It allows users to deposit and withdraw ERC20 tokens from the vault contract and deposit and withdraw shares from the market.
+**/
+contract VaultHelper {
+    using SafeERC20 for IERC20;
+
+    error InsufficientShares();
+    error AddressZero();
+    error MaxDeposit(uint256 amount);
+    error MaxRedeem(uint256 amount);
+
+    IMarket public immutable market;
+    IERC20 public immutable token;
+    IERC4626 public immutable vault;
+
+    /** @dev Constructor
+    @param _vault The address of the ERC4626 vault
+    @param _market The address of the market
+    @param _token The address of the ERC20 token
+    **/
+    constructor(address _vault, address _market, address _token) {
+        vault = IERC4626(_vault);
+        market = IMarket(_market);
+        token = IERC20(_token);
+        token.approve(_vault, type(uint).max);
+        vault.approve(_market, type(uint).max);
+    }
+
+    /**
+    @notice Deposit the associated ERC20 token into the ERC4626 vault and deposit the received shares for recipient.
+    @param recipient The address to receive payment from the escrow
+    @param assets The amount of ERC20 token to be transferred.
+    */
+    function wrapAndDeposit(
+        address recipient,
+        uint assets
+    ) external returns (uint256 shares) {
+        if(recipient == address(0)) revert AddressZero();
+        token.safeTransferFrom(msg.sender, address(this), assets);
+        if (assets > vault.maxDeposit(address(this))) revert MaxDeposit(assets);
+        shares = vault.deposit(assets, address(this));
+        uint256 actualShares = vault.balanceOf(address(this));
+        if (shares > actualShares) revert InsufficientShares();
```

Comment:
this is ineffective since there should never be a mismatch.
i would suggest getting rid of `actualShares` and simply ensuring `shares > 0`.
if you want a more robust defense, could consider a `minSharesOut` parameter.

Reply:
for what it's worth, looking at st-yETH code, it enforces that the final 1e15 shares can never be burned. this makes the cost of an attack incredibly expensive.